blockstore/src/main/java/brooklyn/location/blockstore/AbstractVolumeManager.java

```diff
@@ -73,58 +75,63 @@ public MountedBlockDevice attachAndMountVolume(JcloudsMachineLocation machine, B
     @Override
     public void createFilesystem(AttachedBlockDevice attachedDevice, FilesystemOptions filesystemOptions) {
         JcloudsMachineLocation machine = attachedDevice.getMachine();
-        if (!(machine instanceof SshMachineLocation)) {
-            throw new IllegalStateException("Cannot create filesystem for "+machine+" of type "+machine.getClass().getName()+"; expected "+SshMachineLocation.class.getSimpleName());
-        }
-        
-        String osDeviceName = getOSDeviceName(attachedDevice.getDeviceSuffix());
-        String filesystemType = filesystemOptions.getFilesystemType();
-        LOG.debug("Creating filesystem: device={}; osDeviceName={}, config={}", new Object[]{attachedDevice, osDeviceName, filesystemOptions});
-
-        // NOTE: also adds an entry to fstab so the mount remains available after a reboot.
-        Map<String, ?> flags = MutableMap.of("allocatePTY", true);
-
-        int exitCode = ((SshMachineLocation)machine).execCommands(flags, "Creating filesystem on volume", ImmutableList.of(
-                dontRequireTtyForSudo(),
-                waitForFileCmd(osDeviceName, 60),
-                installPackage(ImmutableMap.of("yum", "e4fsprogs"), null),
-                sudo("/sbin/mkfs -F -t " + filesystemType + " " + osDeviceName)));
-
-        if (exitCode != 0) {
-            throw new RuntimeException(format("Failed to create file system. machine=%s; osDeviceName=%s; filesystemType=%s",
-                    machine, osDeviceName, filesystemType));
+        if (machine instanceof SshMachineLocation) {
+            String osDeviceName = getOSDeviceName(attachedDevice.getDeviceSuffix());
+            String filesystemType = filesystemOptions.getFilesystemType();
+            LOG.debug("Creating filesystem: device={}; osDeviceName={}, config={}", new Object[]{attachedDevice, osDeviceName, filesystemOptions});
+
+            // NOTE: also adds an entry to fstab so the mount remains available after a reboot.
+            Map<String, ?> flags = MutableMap.of("allocatePTY", true);
+
+            int exitCode = ((SshMachineLocation)machine).execCommands(flags, "Creating filesystem on volume", ImmutableList.of(
+                    dontRequireTtyForSudo(),
+                    waitForFileCmd(osDeviceName, 60),
+                    installPackage(ImmutableMap.of("yum", "e4fsprogs"), null),
+                    sudo("/sbin/mkfs -F -t " + filesystemType + " " + osDeviceName)));
+
+            if (exitCode != 0) {
+                throw new RuntimeException(format("Failed to create file system. machine=%s; osDeviceName=%s; filesystemType=%s",
+                        machine, osDeviceName, filesystemType));
+            }
+        } else if (machine instanceof WinRmMachineLocation) {
+            WinRmToolResponse response = ((WinRmMachineLocation)machine).executePsScript("Get-Disk | Where partitionstyle -eq 'raw' | " +
```

Comment:
You could format a disk which you should not format this way.
I prefer passing explicit disk letter.